src/Orleans.Runtime.Legacy/Configuration/BootstrapProviderConfiguration.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Orleans.Providers;
+
+namespace Orleans.Runtime.Configuration
+{
+    public static class BootstrapProviderConfiguration
+    {
+        public const string BOOTSTRAP_PROVIDER_CATEGORY_NAME = "Bootstrap";
+
+        /// <summary>
+        /// Registers a given type of <typeparamref name="T"/> where <typeparamref name="T"/> is bootstrap provider
+        /// </summary>
+        /// <typeparam name="T">Non-abstract type which implements <see cref="IBootstrapProvider"/> interface</typeparam>
+        /// <param name="providerName">Name of the bootstrap provider</param>
+        /// <param name="properties">Properties that will be passed to bootstrap provider upon initialization</param>
+        public static void RegisterBootstrapProvider<T>(this GlobalConfiguration config, string providerName, IDictionary<string, string> properties = null) where T : IBootstrapProvider
```

Comment:
ohhhh ignore my earlier comment on `GlobalConfiguration`, you moved those methods here.

Reply:
but on a side note. if `GlobalConfiguration` is going to be moved to `Core.Legacy` or `Runtime.Legacy` package, is there still value to keep those as extension methods? Might just keep them as `GlobalConfiguration` methods as before.

Or is this a intermedia step for moving boostrap provider to legacy project

Reply:
This depends on how we choose to move the legacy configs, if we do it all at once then this is probably redundant.  I wrote this when the plan was to move things in chunks.